src/test/unit/model/indexing/lvalue_varmat_test.cpp

```diff
@@ -0,0 +1,760 @@
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(4));
+  test_throw_out_of_range(x, index_list(index_min_max(0, 3)), y);
+  test_throw_out_of_range(x, index_list(index_min_max(1, 8)), y);
+}
+
+TEST_F(VarAssign, positive_minmax_vec) {
+  test_positive_minmax_varvector<Eigen::VectorXd>();
+}
+
+TEST_F(VarAssign, positive_minmax_rowvec) {
+  test_positive_minmax_varvector<Eigen::RowVectorXd>();
+}
+
+template <typename Vec>
+void test_negative_minmax_varvector() {
+  using stan::math::sum;
+  using stan::math::var_value;
+  using stan::model::test::check_vector_adjs;
+  using stan::model::test::generate_linear_var_vector;
+  auto x = generate_linear_var_vector<Vec>(5);
+  Vec x_val = x.val();
+  auto y = generate_linear_var_vector<Vec>(4, 10);
+
+  assign(x, index_list(index_min_max(4, 1)), y);
+  EXPECT_FLOAT_EQ(x.val()(0), 13);
+  EXPECT_FLOAT_EQ(x.val()(1), 12);
+  EXPECT_FLOAT_EQ(x.val()(2), 11);
+  EXPECT_FLOAT_EQ(x.val()(3), 10);
+  EXPECT_FLOAT_EQ(x.val()(4), 4);
+  sum(x).grad();
+  EXPECT_MATRIX_EQ(x.val(), x_val);
+  auto check_i = [](int i) { return (i > 3); };
+  check_vector_adjs(check_i, x, "lhs");
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(4));
+  test_throw_out_of_range(x, index_list(index_min_max(3, 0)), y);
+  test_throw_out_of_range(x, index_list(index_min_max(8, 1)), y);
+}
+
+TEST_F(VarAssign, negative_minmax_vec) {
+  test_negative_minmax_varvector<Eigen::VectorXd>();
+}
+
+TEST_F(VarAssign, negative_minmax_rowvec) {
+  test_negative_minmax_varvector<Eigen::RowVectorXd>();
+}
+
+template <typename Vec>
+void test_uni_uni_vec_eigvec() {
+  using stan::math::sum;
+  using stan::math::var;
+  using stan::math::var_value;
+
+  Vec xs0_val(3);
+  xs0_val << 0.0, 0.1, 0.2;
+
+  Vec xs1_val(3);
+  xs1_val << 1.0, 1.1, 1.2;
+
+  var_value<Vec> xs0(xs0_val);
+  var_value<Vec> xs1(xs1_val);
+  vector<var_value<Vec>> xs;
+  xs.push_back(xs0);
+  xs.push_back(xs1);
+
+  var y = 15;
+  assign(xs, index_list(index_uni(2), index_uni(3)), y);
+  EXPECT_FLOAT_EQ(y.val(), xs[1].val()(2));
+
+  (sum(xs[0]) + sum(xs[1])).grad();
+  EXPECT_FLOAT_EQ(y.adj(), 1);
+
+  for (Eigen::Index i = 0; i < xs[0].size(); ++i) {
+    EXPECT_FLOAT_EQ(xs[0].val()(i), xs0_val(i));
+    EXPECT_FLOAT_EQ(xs[0].adj()(i), 1);
+  }
+
+  for (Eigen::Index i = 0; i < xs[0].size(); ++i) {
+    EXPECT_FLOAT_EQ(xs[1].val()(i), xs1_val(i));
+    if (i == 2) {
+      EXPECT_FLOAT_EQ(xs[1].adj()(i), 0);
+    } else {
+      EXPECT_FLOAT_EQ(xs[1].adj()(i), 1);
+    }
+  }
+
+  test_throw_out_of_range(xs, index_list(index_uni(0), index_uni(3)), y);
+  test_throw_out_of_range(xs, index_list(index_uni(2), index_uni(0)), y);
+  test_throw_out_of_range(xs, index_list(index_uni(10), index_uni(3)), y);
+  test_throw_out_of_range(xs, index_list(index_uni(2), index_uni(10)), y);
+}
+
+TEST_F(VarAssign, uni_uni_std_vecvec) {
+  test_uni_uni_vec_eigvec<Eigen::VectorXd>();
+}
+
+TEST_F(VarAssign, uni_uni_std_vecrowvec) {
+  test_uni_uni_vec_eigvec<Eigen::RowVectorXd>();
+}
+
+// Uni Assigns
```

Comment:
Is that comment supposed to tell me anything?

Reply:
Nah not really